<head>
    <title>mCED</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="shortcut icon" href="static/images/icon.png" type="image/x-icon">
    <link rel="icon" href="static/images/icon.png" type="image/x-icon">
    <link rel=stylesheet type=text/css href="static/typeaheadjs.css">
    <link rel=stylesheet type=text/css href="static/css/font-awesome.min.css">
    <script type="text/javascript" src="static/jquery-1.11.1.min.js"></script>
    <script type="text/javascript" src="static/bootstrap.min.js"></script>
    <script type="text/javascript" src="static/typeahead.bundle.min.js"></script>

    <script type="text/javascript" src="static/jquery.tablesorter.min.js"></script>
    <script type="text/javascript" src="static/jquery.tablesorter.pager.js"></script>
    <link rel=stylesheet type=text/css href="static/theme.default.css">
    <script type="text/javascript" src="static/underscore-min.js"></script>

    <link rel=stylesheet type=text/css href="static/bootstrap.min.css">
    <link rel=stylesheet type=text/css href="static/style.css">
    <script type="text/javascript" src="static/d3.v3.min.js"></script>
    <script type="text/javascript" src="static/index.js"></script>
    <script type="text/javascript" src="static/exac.js"></script>
    <script type="text/javascript">
        number_of_samples = 2123;
        release_number = 1.0;
        number_of_samples_full = 2123;
        $(document).ready(function() {
            $('.number_samples').html(Number(number_of_samples).toLocaleString('en'));
            $('.number_samples_full').html(Number(number_of_samples_full).toLocaleString('en'));
            $('.release_number').html(Number(release_number).toLocaleString('en'));
        });
        $(function() {
            var bestPictures = new Bloodhound({
              datumTokenizer: Bloodhound.tokenizers.obj.whitespace('value'),
              queryTokenizer: Bloodhound.tokenizers.whitespace,
              remote: '/autocomplete/%QUERY'
            });

            bestPictures.initialize();

            $('.awesomebar').typeahead(
                {
                    autoselect: true,
                },
                {
                    name: 'best-pictures',
                    displayKey: 'value',
                    source: bestPictures.ttAdapter(),
                }
            );
            $('.awesomebar').bind('typeahead:selected', function(obj, datum) {
                window.location.href = '/awesome?query=' + datum.value;
            });
        });
    </script>
</head>
<body>
<nav class="navbar navbar-default" role="navigation" style="background: #103173;">
    <div class="container-fluid">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle pull-right" data-toggle="collapse" data-target="#navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            
            <a class="navbar-brand" href="index.html" style="color: white; font-weight: bold; float: left; font-size: 15px;">mCED</a>
            
            <div class="col-xs-5" id="navbar_form_container">
                <form action="gene.php" class="navbar-form" role="search">
                    <div class="form-group" id="navbar-awesomebar">
                        <input type="submit" style="display: none;"/>
                        <input id="navbar-searchbox-input" name="query" class="form-control awesomebar" type="text" placeholder="Gene or Ensembl ID"/>
                        <input type="submit" style="position: absolute; left: -9999px"/>
                    </div>
                </form>
            </div>
        </div>
        <!-- Collect the nav links, forms, and other content for toggling -->
        <div class="collapse navbar-collapse" id="navbar-collapse">
            <ul class="nav navbar-nav navbar-right">
                <li><a href="about.html" style="color: white; font-weight: bold;">About</a></li>
                <li><a href="downloads.html" style="color: white; font-weight: bold;">Downloads</a></li>
                <li><a href="terms.html" style="color: white; font-weight: bold;">Terms</a></li>
                <li><a href="contact.html" style="color: white; font-weight: bold;">Contact</a></li>
                <li><a href="faq.html" style="color: white; font-weight: bold;">FAQ</a></li>
            </ul>
        </div><!-- /.navbar-collapse -->
    </div><!-- /.container-fluid -->
</nav>


    <div class="container">
        <h1>Terms and Data Information</h1>
        <div class="row" style="font-size: 16px;">
            <h4>Terms of use</h4>
            <p>
	    The data presented here are released for the broader research community's benefit. You are welcome to freely download and explore the data. We encourage utilization and publication of frequency data for specific targeted sets of variants, such as analyzing candidate causal variants found in rare disease patients. However, we request refraining from publishing global (genome-wide) analyses of this data or large gene sets until after our papers are published, which is anticipated in 2025.
            </p>
            <h4>Citation in publications</h4>
            <p>
	    Please cite our bioRxiv preprint for usage of the data from the Macaque Capture and Whole Exome Database (mCED) browser.
            </p>
            <h4>Data Generation</h4>
            <p>
	    The sequencing reads were mapped to the rhesus reference genome assembly (Mmul_8.0.1 or Mmul_10) using BWA mem. Single nucleotide variants (SNVs) and short insertion/deletions (indels) were called following the GATK pipeline. Variants called based on the Mmul_8.0.1 genome assembly were subsequently mapped to the Mmul_10 assembly using the UCSC genome browser program liftOver. Any sites with mismatched reference nucleotides were corrected using bcftools.
            </p>
            <p>
	    The sequenced variants underwent quality control and filtering based on the following criteria:
1. Variants labeled as "PASS" by GATK, meeting specific filtering conditions: QD < 5.0, QUAL < 30.0, FS > 15.0, MQ < 50.0, MQRankSum < -12.5, and ReadPosRankSum < -8.0.

2. Variants that remained dimorphic after assigning missing variants according to the following criteria: a) variants not called; b) heterozygous variant calls with allelic imbalance (AB>0.8 or AB<0.2); c) variant GQ per sample (from GATK) < 20; d) genotypes supported by less than 10 reads (DP < 10).

3. Sites where more than two distinct alleles were detected in a single subject were filtered out.
            </p>
        </div>
    </div>

</body>
